### DISCUSSION TITLE
failing or missing response from https://IP:6443/apis/external.metrics.k8s.io/v1beta1

### DISCUSSION THREAD

**Discussion by mo-saeed** [kedacore/keda#1835] [Q&A / Need Help] [answered]
### Report

Keda was installed successfully without issues and both pods have clear logs with no errors 

but with this command i see `kubectl get apiservice v1beta1.external.metrics.k8s.io -o yaml`

I get

```
conditions:
  - lastTransitionTime: "2021-05-26T13:59:23Z"
    message: 'failing or missing response from https://KEDA_METRICS_SERVER_IP:6443/apis/external.metrics.k8s.io/v1beta1:
      Get "https://KEDA_METRICS_SERVER_IP:6443/apis/external.metrics.k8s.io/v1beta1": net/http:
      request canceled while waiting for connection (Client.Timeout exceeded while
      awaiting headers)'
    reason: FailedDiscoveryCheck
    status: "False"
    type: Available
```
my cluster is EKS with kubernetes v1.19

I am not sure what is the issue here, I saw this https://keda.sh/docs/2.2/troubleshooting/ but my EKS firewall looks fine.

Can you please help ?

Also is it possible to use existing metrics-server installation, instead of using keda-operator-metrics-server.

Thanks

### Expected Behavior

I should expect SUCCESS status with this `kubectl get apiservice v1beta1.external.metrics.k8s.io -o yaml`

### Actual Behavior

I get status FAILED with this `kubectl get apiservice v1beta1.external.metrics.k8s.io -o yaml`

### Steps to Reproduce the Problem

1. install keda with helm in EKS cluster using helm (v2.2)
2. run `kubectl get apiservice v1beta1.external.metrics.k8s.io -o yaml`

### Logs from KEDA operator

```
I0526 13:59:33.175579       1 request.go:655] Throttling request took 1.046797067s, request: GET:https://172.20.0.1:443/apis/vpcresources.k8s.aws/v1beta1?timeout=32s
2021-05-26T13:59:35.879Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
2021-05-26T13:59:35.881Z	INFO	controllers.ScaledObject	Running on Kubernetes 1.19+	{"version": "v1.19.6-eks-49a6c0"}
2021-05-26T13:59:35.881Z	INFO	setup	Starting manager
2021-05-26T13:59:35.881Z	INFO	setup	KEDA Version: 2.2.0
2021-05-26T13:59:35.881Z	INFO	setup	Git Commit: 6dc4c3524ba6278276cd1338cc7e4e9a7dd88963
2021-05-26T13:59:35.882Z	INFO	setup	Go Version: go1.15.6
2021-05-26T13:59:35.882Z	INFO	setup	Go OS/Arch: linux/amd64
I0526 13:59:35.882269       1 leaderelection.go:243] attempting to acquire leader lease kube-infrastructure/operator.keda.sh...
2021-05-26T13:59:35.882Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I0526 13:59:53.300932       1 leaderelection.go:253] successfully acquired lease kube-infrastructure/operator.keda.sh
2021-05-26T13:59:53.301Z	INFO	controller	Starting EventSource	{"reconcilerGroup": "keda.sh", "reconcilerKind": "ClusterTriggerAuthentication", "controller": "clustertriggerauthentication", "source": "kind source: /, Kind="}
```

### KEDA Version

2.2.0

### Kubernetes Version

1.19

### Platform

Amazon Web Services

### Scaler Details

Azure Service Bus

**Comment by tomkerkhove** [comment]
Hm odd, this should work indeed. Are you using network policies?

Do you have any ideas @zroubalik?

> Also is it possible to use existing metrics-server installation, instead of using keda-operator-metrics-server.

Unfortunately this is not possible as you can only have 1 metric server in a cluster

**Reply by coderanger** [reply]
More specifically the HPA system uses hardcoded API IDs for each of the metrics types. You can have as many metrics servers as you want but only one thing can listen on a given API ID. That said, metrics-server only listens on the metrics.k8s.io ID and KEDA listens on external.metrics.k8s.io so the two shouldn't conflict. What you can't do is use two things on external.metrics.k8s.io together, like KEDA and prometheus-adapter.

**Comment by mo-saeed** [comment, marked as answer]
@tomkerkhove Thanks a lot for the hint, yes it was missing network policy.

**Reply by tomkerkhove** [reply]
Happy to help!

**Reply by agconti** [reply]
What was the policy you were missing?

**Reply by emmanuelsosa** [reply]
Probably 6443 from the EKS Cluster SG to the Nodes SG.

**Reply by tomkerkhove** [reply]
Is there anything we can add to our troubleshooting guide (https://keda.sh/docs/2.7/troubleshooting/)?

**Reply by stv-io** [reply]
As someone who found this information useful, coming, from google, terraform + eks - I would say this helped me, and I'd suggest
 - documenting this in the troubleshooting guide, clearly stating the tcp6443 from cluster to nodes is permitted (SG and/or network policy)
 - possibly, go one step further, and propose this in the EKS module in terraform as another of the `node_security_group_recommended_rules ` in https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/node_groups.tf#L109

(it's still configurable via the below)

```
  create_node_security_group = true
  node_security_group_additional_rules = {
    keda_metrics_server_access = {
      description                   = "Cluster to nodes keda metrics serve"
      protocol                      = "tcp"
      from_port                     = 6443
      to_port                       = 6443
      type                          = "ingress"
      source_cluster_security_group = true
    }
...
```
ref:

```
module "eks" {
  source                               = "terraform-aws-modules/eks/aws"
  version                              = "19.5.1"
```

**Reply by zroubalik** [reply]
@stv-io are you willing to contribute this to our docs please? In the troubleshooting section: https://keda.sh/docs/2.9/troubleshooting/

**Reply by stv-io** [reply]
Sure - https://github.com/kedacore/keda-docs/pull/1047 - hope I did things as expected
